pkg/cmd/util/clientcmd/clientcmd.go

```diff
@@ -76,6 +87,33 @@ func EnvVars(host string, caData []byte, insecure bool, bearerTokenFile string)
 func (cfg *Config) bindEnv() error {
 	var err error
 
+	// callers may not use the config file if they have specified a master directly
+	_, masterSet := util.GetEnv("OPENSHIFT_MASTER")
+	specifiedMaster := masterSet || cfg.MasterAddr.Provided
+
+	if cfg.clientConfig != nil && !specifiedMaster {
+		clientConfig, err := cfg.clientConfig.ClientConfig()
+		if err != nil {
+			return err
+		}
+		cfg.CommonConfig = *clientConfig
+		cfg.Namespace, _, err = cfg.clientConfig.Namespace()
+		if err != nil {
+			return err
+		}
+		if !cfg.MasterAddr.Provided {
+			cfg.MasterAddr.Set(cfg.CommonConfig.Host)
+		}
+		if !cfg.KubernetesAddr.Provided {
+			cfg.KubernetesAddr.Set(cfg.CommonConfig.Host)
+		}
+		if value, ok := util.GetEnv("OPENSHIFT_INSECURE"); ok && len(value) != 0 {
```

Comment:
I was fine until here.  Why does the env-var need to overwrite the kubeconfig file value?  They haven't specified `OPENSHIFT_MASTER`, so we know that this is a new master using a new image.  It seems like we could reasonably ignore the envvars entirely.